network/server_test.go

```diff
@@ -14,26 +15,174 @@ func TestDialLifecycle(t *testing.T) {
 	// we should remove discovery protocol and just pipe values to the dialqueue to see how
 	// it behaves
 
-	cfg := func(c *Config) {
-		c.MaxPeers = 1
-		// c.NoDiscover = true
+	type Request struct {
+		from      uint
+		to        uint
+		isSuccess bool
+	}
+	type Expected struct {
+		numPeers int64
+	}
+
+	tests := []struct {
+		name     string
+		cfgs     []func(*Config)
+		requests []Request
+		expected []Expected
+	}{
+		{
+			name: "server 1 should reject incoming connection request from server 3 due to limit",
+			cfgs: []func(*Config){
+				func(c *Config) {
```

Comment:
Since this config is repeated a lot, could you create something like
```
defConfig := func(c *Config) {
   c.MaxPeers = 1
   c.NoDiscover = true
}
```
and then use it instead